homeassistant/components/switch/flux.py

```diff
@@ -140,8 +140,11 @@ def turn_on(self, **kwargs):
         if not self._state:  # make initial update
             self.flux_update()
         self._state = True
-        self.unsub_tracker = track_time_change(self.hass, self.flux_update,
-                                               second=[0, 30])
+
+        if self.unsub_tracker is None:
```

Comment:
I would expect instead of doing this check to just add an `if self._state: return` to the beginning.

I also guess that `self._state` is not necessary, it can be dereived from `self.unsub_tracker is not None`